File: Core/Workspace/Typescript.legacy/Site/wwwroot/Core/Workspace/Protocol/Sync/SyncResponse.ts
namespace Allors.Data {
    export interface SyncResponse extends Response {
        accessControls?: string[][];
        objects: SyncResponseObject[];
    }

    export interface SyncResponseObject {
        i: string;
        t: string;
        v: string;
        a?: string;
        d?: string;
        r?: SyncResponseRole[];
    }

    export interface SyncResponseRole {
        t: string;
        v: string;
    }
}

File: Platform/Protocol/Allors.Protocol/Api/Sync/SyncResponse.cs
// <copyright file="SyncResponse.cs" company="Allors bv">
// Copyright (c) Allors bv. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Remote.Sync
{
    public class SyncResponse
    {
        /// <summary>
        /// AccessControls
        /// </summary>
        public string[][] accessControls { get; set; }

        /// <summary>
        /// Objects
        /// </summary>
        public SyncResponseObject[] objects { get; set; }
    }
}
